src/osd/OSD.cc

```diff
@@ -4154,6 +4154,8 @@ int OSD::shutdown()
 {
   if (cct->_conf->osd_fast_shutdown) {
     derr << "*** Immediate shutdown (osd_fast_shutdown=true) ***" << dendl;
```

Comment:
A quick suggestion: log the three relevant values: osd_fast_shutdown, osd_fast_shutdown_notify_mon, and osd_mon_shutdown_timeout here. Perhaps even shift this outside of the osd_fast_shutdown clause.

Reply:
When reading the logs, it would be helpful to quickly know which shutdown scenario applies. This could also help catch instances where settings were modified between the logged issue and the sosreport capture.

That said, I don't expect these settings will be adjusted frequently. There is also enough logging elsewhere to deduce which settings were applied at run-time. So, I'm just leaving this as a suggested improvement.